Makefile.Common

```diff
@@ -22,12 +22,34 @@ GOCMD?= go
 GOTEST=$(GOCMD) test
 GOOS=$(shell $(GOCMD) env GOOS)
 GOARCH=$(shell $(GOCMD) env GOARCH)
-ADDLICENCESE= addlicense
-MDLINKCHECK=markdown-link-check
-MISSPELL=misspell -error
-MISSPELL_CORRECTION=misspell -w
-LINT=golangci-lint
-IMPI=impi
+
+# In order to help reduce toil related to managing tooling for the open telemetry collector
+# this section of the makefile looks at only requiring command definitions to be defined
+# as part of $(TOOLS_MOD_DIR)/tools.go, following the existing practice.
+# Modifying the tools' `go.mod` file will trigger a rebuild of the tools to help
+# ensure that all contributors are using the most recent version to make builds repeatable everywhere.
+TOOLS_MOD_DIR    := $(PWD)/internal/tools
+TOOLS_MOD_REGEX  := "\s+_\s+\".*\""
+TOOLS_PKG_NAMES  := $(shell grep -E $(TOOLS_MOD_REGEX) < $(TOOLS_MOD_DIR)/tools.go | tr -d " _\"")
+TOOLS_BIN_DIR    := $(PWD)/bin
```

Comment:
I would suggest `.tools` instead of `bin`.  This is the location used by the [Go](https://github.com/open-telemetry/opentelemetry-go/blob/e8c6e4517819c57978a7eb2e14d24b9cf0eb7a29/Makefile#L33-L39) and [Go contrib](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/Makefile#L39-L45) repos and aligns with the convention that directories starting with `.` can be ignored in most cases.

Reply:
make sense to me, I was using bin since it was ignored and the nix default naming.